src/controllers/dynakube/dynakube_controller.go

```diff
@@ -108,44 +110,58 @@ type Controller struct {
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (controller *Controller) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling DynaKube", "namespace", request.Namespace, "name", request.Name)
-	requeueAfter := defaultUpdateInterval
 
-	dynakube, err := controller.getDynakubeOrUnmap(ctx, request.Name, request.Namespace)
+	dynaKube, err := controller.getDynakubeOrUnmap(ctx, request.Name, request.Namespace)
 	if err != nil {
 		return reconcile.Result{}, err
-	} else if dynakube == nil {
+	} else if dynaKube == nil {
 		return reconcile.Result{}, nil
 	}
 
-	oldStatus := *dynakube.Status.DeepCopy()
-	err = controller.reconcileDynaKube(ctx, dynakube)
+	return controller.reconcile(ctx, dynaKube)
+}
 
-	if err != nil {
-		requeueAfter = errorUpdateInterval
-
-		var serverErr dtclient.ServerError
-		isServerError := errors.As(err, &serverErr)
-		if isServerError && (serverErr.Code == http.StatusTooManyRequests || serverErr.Code == http.StatusServiceUnavailable) {
-			// should we set the phase to error ?
-			log.Info("server is unavailable or request limit reached! trying again in one minute")
-			return reconcile.Result{RequeueAfter: requeueAfter}, nil
-		}
-		dynakube.Status.SetPhase(dynatracestatus.Error)
-		log.Error(err, "error reconciling DynaKube", "namespace", dynakube.Namespace, "name", dynakube.Name)
-	} else {
-		dynakube.Status.SetPhase(controller.determineDynaKubePhase(dynakube))
+func (controller *Controller) setRequeueAfterIfNewIsShorter(requeueAfter time.Duration) {
+	if controller.requeueAfter > requeueAfter {
+		controller.requeueAfter = requeueAfter
+	}
+}
+
+func (controller *Controller) reconcile(ctx context.Context, dynaKube *dynatracev1beta1.DynaKube) (reconcile.Result, error) {
+	oldStatus := *dynaKube.Status.DeepCopy()
+
+	controller.requeueAfter = defaultUpdateInterval
+
+	err := controller.reconcileDynaKube(ctx, dynaKube)
+
+	var serverErr dtclient.ServerError
+	switch {
+	case errors.As(err, &serverErr) && (serverErr.Code == http.StatusTooManyRequests || serverErr.Code == http.StatusServiceUnavailable):
```

Comment:
this is proper custom error checking 👍

Reply:
nitpick: we could have a little abstraction for `errors.As(err, &serverErr) && (serverErr.Code == http.StatusTooManyRequests || serverErr.Code == http.StatusServiceUnavailable)`, like `isDynatraceAPIUnavailable(err)` or something like that